### PR TITLE
Feat/#242 error msg

### PR DESCRIPTION
**Pull request by ciaranschutte**
only two files edited for error message. these are commented. the rest are TS changes.
if you want me to split them into 2 PR's let me know. it was easier to fix them while TS was yelling at me.

- adds upload error for ethics and sign and submit
- adds TS rule for `noUsuedLocals`, fixes current files to pass rule

### CODE REVIEW

File: components/pages/Applications/ApplicationForm/Forms/Signature.tsx [hunk 1]
```diff
@@ -60,6 +60,7 @@ const Signature = ({
   const dismissModal = () => setModalVisible(false);
 
   const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submissionError, setSubmissionError] = useState(false);
```
Comment (ciaranschutte) [inline components/pages/Applications/ApplicationForm/Forms/Signature.tsx:63]: add submission error for Signature failed upload

File: components/pages/Applications/ApplicationForm/Forms/EthicsLetter/UploadsTable.tsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ const UploadsTable = ({
   };
 
   const submitFile = (file?: File) => {
+    setLetterError(false);
```
Comment (ciaranschutte) [inline components/pages/Applications/ApplicationForm/Forms/EthicsLetter/UploadsTable.tsx:88]: add additional error setting for failed upload in ethics

**Comment by ciaranschutte** [comment]
https://github.com/icgc-argo/dac-api/pull/153

File: components/pages/Applications/ApplicationForm/Forms/ProjectInfo/PublicationURLs.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { FormFieldType } from '../types';
 import { getMin, isRequired } from '../validations';
 import { StaticPublications } from 'components/pages/Applications/PDF/StaticProjectInfo';
 
-const ID = 'publicationsURLs';
+//const ID = 'publicationsURLs';
```
Comment (samrichca) [inline components/pages/Applications/ApplicationForm/Forms/ProjectInfo/PublicationURLs.tsx:17]: is this an unused variable being saved for later?

File: global/utils/egoTokenUtils.tsx [hunk 4]
```diff
@@ -6,10 +6,10 @@ import { EgoJwtData, UserWithId } from 'global/types';
 
 const TokenUtils = createEgoUtils(getConfig().NEXT_PUBLIC_EGO_PUBLIC_KEY);
 
-type PermissionScopeObj = {
+/* type PermissionScopeObj = {
```
Comment (samrichca) [inline global/utils/egoTokenUtils.tsx:9]: same here is this being saved for later?

File: components/pages/Applications/ApplicationForm/Forms/ProjectInfo/PublicationURLs.tsx (hunk 3 above)
Comment (ciaranschutte) [reply components/pages/Applications/ApplicationForm/Forms/ProjectInfo/PublicationURLs.tsx:17]: not sure if others had these here for something. commented out so the new TS `noUnusedLocals` doesn't complain.

File: global/utils/egoTokenUtils.tsx (hunk 4 above)
Comment (ciaranschutte) [reply global/utils/egoTokenUtils.tsx:9]: not sure if others had these here for something. commented out so the new TS `noUnusedLocals` doesn't complain.

**Review by samrichca — APPROVED** [review]

**Pull request merged**